spec.md

```diff
@@ -5,76 +5,97 @@ keywords: registry, on-prem, images, tags, repository, distribution, api, advanc
 
-An implementation is not compliant if it fails to satisfy one or more of the MUST, MUST NOT, REQUIRED, SHALL, or SHALL NOT requirements for the protocols it implements.
-An implementation is compliant if it satisfies all the MUST, MUST NOT, REQUIRED, SHALL, and SHALL NOT requirements for the protocols it implements.
+### Historical Context
 
-## Historical Context
+The spec is based on the specification for the [Docker Registry HTTP API V2 protocol](https://github.com/docker/distribution/blob/5cb406d511b7b9163bff9b6439072e4892e5ae3b/docs/spec/api.md).
 
 For relevant details and a history leading up to this specification, please see the following issues:
 
 - [moby/moby#8093](https://github.com/moby/moby/issues/8093)
 - [moby/moby#9015](https://github.com/moby/moby/issues/9015)
 - [docker/docker-registry#612](https://github.com/docker/docker-registry/issues/612)
 
-<!--- TODO: add relevant background information here --->
+### Definitions
+
+Several terms are used frequently in this document and warrant basic definitions:
+
+- **Registry**: a HTTP service which implements this spec
+- **Client**: a tool that communicates with registries over HTTP
+- **Push**: the act of uploading content to a registry
+- **Pull**: the act of downloading content from a registry
+- **Artifact**: a single piece of content, made up of a manifest and one or more layers
+- **Manifest**: a JSON document which defines an artifact
+- **Layer**: a single part of all the parts which comprise an artifact
+- **Config**: a special layer defined at the top of a manifest containing artifact metadata
+- **Blob**: a single binary content stored in a registry
+- **Digest**: a unique blob identifier, calculated from the blob's content
+- **Content**: a general term for content that can be downloaded from a registry (manifest or blob), and are addressable by a digest
+
+## Conformance
+
+### Notational Conventions
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" are to be interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119) (Bradner, S., "Key words for use in RFCs to Indicate Requirement Levels", BCP 14, RFC 2119, March 1997).
+
+### Minimum Requirements
+
+For a registry to be considered fully conformant against this specification, it must implement the HTTP endpoints required by each of the four (4) major workflow categories:
+
+1. **Pull** (REQUIRED) - Ability to fetch content from a registry
+2. **Push** - Ability to publish content to a registry
+3. **Content Discovery** - Ability to list or otherwise query the content stored in a registry
+4. **Content Management** - Ability to control the full life-cycle of the content stored in a registry
+
+At a bare minimum, registries claiming to be "OCI-Compliant" MUST support all facets of the pull workflow.
```

Comment:
Suggest: `To be considered "OCI-Compliant" to this distribution specification, the registry MUST support all facets of the pull workflow.`

also needs link..

Reply:
link to what?

Reply:
ive just removed it